usecases/schema/add_property.go

```diff
@@ -32,6 +30,13 @@ func (m *Manager) AddClassProperty(ctx context.Context, principal *models.Princi
 		return err
 	}
 
+	if property.Name == "" {
+		return fmt.Errorf("property must contain name")
+	}
+	if property.DataType == nil {
```

Comment:
just a suggestion maybe here would make sense to change this condition to if `len(property.DataType) == 0`? that way we would catch `== nil` and `empty array` cases, but I don't know if `empty array` will ever occur, will it? if not then please discard my comment.

Reply:
Both cases are accounted for 🙂 

If you send a request body like this:
```json
{
  "name": "newProp"
}
```

The response is 
```json
{
  "error": [
    {
      "message": "property must contain dataType"
    }
  ]
}
```

and if the request looks like:
```json
{
  "name": "newProp",
  "dataType": []
}
```

the response:
```json
{
  "error": [
    {
      "message": "property 'newProp': invalid dataType: dataType must have at least one element"
    }
  ]
}
```